### PR TITLE
fixes #5617, 'macros.lineInfo=' addition

### LINKED ISSUE #5617: Ability to set a NimNode's lineinfo
The only way to set this information is via ``newNimNode``, but a lot of the time I have a node that was created via ``newCall`` or some other convenience function. These functions don't allow me to specify the line info information so it would be nice to be able to assign it after the fact.

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -432,6 +432,12 @@ proc getLine(arg: NimNode): int {.magic: "NLineInfo", noSideEffect.}
 proc getColumn(arg: NimNode): int {.magic: "NLineInfo", noSideEffect.}
 proc getFile(arg: NimNode): string {.magic: "NLineInfo", noSideEffect.}
 
+proc lineInfo*(arg: NimNode, info: NimNode) {.magic: "NLineInfo", noSideEffect.}
```
Comment (Araq) [inline lib/core/macros.nim]: This shouldn't be available, only offer the proper setter syntax.

File: tests/macros/tlineinfo.nim [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# issue #5617, feature request
+# Ability to set a NimNode's lineinfo
+import macros
+
+type
+  Test = object
+
+macro mixer(n: typed): untyped =
+  let x = newIdentNode("echo")
+  x.lineInfo = n
```
Comment (dom96) [inline tests/macros/tlineinfo.nim]: This should be ``x.lineInfo = n.lineInfo``, it doesn't make sense otherwise!

Comment (jangko) [reply tests/macros/tlineinfo.nim]: there is no way to set `info.filename` from outside VM. We can getFile from a node but we cannot use setFile to a node.
both `macros.newNimNode` and `macros.error` use another node to obtain fileinfo.

Comment (dom96) [reply tests/macros/tlineinfo.nim]: Then you need to rename this procedure to something else.

Comment (jangko) [reply tests/macros/tlineinfo.nim]: I think `copyLineInfo` would be a nice name

**Pull request merged**